services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -54,6 +59,61 @@ export const DynamicField = ({ name, label, ...props }: Props) => {
     setDisplayValues(newDisplayValues);
   };
 
+  // delete selected record from DB
+  const deleteRecord = async (selectedRecord: EntityShape) => {
+    if (userIsStateUser || userIsStateRep) {
+      const reportKeys = {
+        state: state,
+        id: report?.id,
+      };
+
+      let filteredEntities = {};
+      if (name === "plans") {
+        filteredEntities = {
+          [name]: form
+            .getValues()
+            .plans.filter((plan: AnyObject) => plan.id !== selectedRecord.id),
+        };
+      } else if (name === "bssEntities") {
+        filteredEntities = {
+          [name]: form
+            .getValues()
+            .bssEntities.filter(
+              (bssEntity: AnyObject) => bssEntity.id !== selectedRecord.id
+            ),
+        };
+      }
+
+      const dataToWrite = {
+        status: ReportStatus.IN_PROGRESS,
+        lastAlteredBy: full_name,
+        fieldData: filteredEntities,
+      };
+
+      // delete related sanctions and quality measures
+      if (report?.fieldData.sanctions) {
```

Comment:
does we need a question mark before `sanctions`? (does it throw an error if we delete a plan when there are no sanctions?)